packages/prepo-utils/__tests__/utils.test.ts

```diff
@@ -37,3 +38,45 @@ describe('prepo-utils', () => {
     expect(formattedNumber3).toEqual('$123M')
   })
 })
+
+describe('parseUnits function', () => {
+  it('returns undefined if decimals is undefined', () => {
+    const output = parseUnits('100', undefined)
+    expect(output).toBe(undefined)
+  })
+
+  it('returns zero BigNumber if string is empty', () => {
+    const output = parseUnits('', 6)
+    expect(output.eq(BigNumber.from(0))).toBe(true)
+  })
+
+  it('returns 100000000 if value is 100 in number format and decimals is 6', () => {
+    const output = parseUnits(100, 6)
+    expect(output.eq(BigNumber.from(100000000))).toBe(true)
+  })
+
+  it('returns 100000000 if value is 100 in string format and decimals is 6', () => {
+    const output = parseUnits('100', 6)
+    expect(output.eq(BigNumber.from(100000000))).toBe(true)
+  })
+
+  it('returns 123456 if value is 0.1234567890 and decimals is 6', () => {
+    const output = parseUnits('0.1234567890', 6)
+    expect(output.eq(BigNumber.from(123456))).toBe(true)
+  })
+
+  it('returns 123456789000000000 if value is 0.1234567890 and decimals is 18', () => {
+    const output = parseUnits('0.1234567890', 18)
+    expect(output.eq(BigNumber.from('123456789000000000'))).toBe(true)
+  })
+
+  it('returns undefined if fraction part contains non-number character', () => {
+    const output = parseUnits('0.123456789123456789123456789abc', 18)
+    expect(output).toBe(undefined)
+  })
+
+  it('returns undefined without throwing error if value is invalid number', () => {
```

Comment:
(not throwing error is implied)
```suggestion
  it('returns undefined if value is invalid number', () => {
```